src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs

```diff
@@ -189,6 +166,49 @@ internal void ValidateCollection()
                 }
             }
         }
+
+        void ICollection.CopyTo(Array array, int index) => NonGenericList.CopyTo(array, index);
+
+        int IList.Add(object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            return NonGenericList.Add(value);
+        }
+
+        bool IList.Contains(object value) => NonGenericList.Contains(value);
+
+        int IList.IndexOf(object value) => NonGenericList.IndexOf(value);
+
+        void IList.Insert(int index, object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            NonGenericList.Insert(index, value);
+        }
+
+        void IList.Remove(object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            bool removed = _list.Remove(value);
+
+            // This throws on full framework, so it will also throw here.
+            if (!removed)
+            {
+                throw new ArgumentException(SR.Arg_RemoveArgNotFound);
+            }
+        }
+
+        private IList NonGenericList => _list;
```

Comment:
We need to call some of the explicitly implemented non-generic collection interface members on `List<T>` and having this property is a convenient way to do that. Otherwise, we'd have to case the `List<T>` to the non-generic interfaces at the callsites, e.g.:

```csharp
((ICollection)_list).CopyTo(array, index);
```

We did the same thing in `X509CertificateCollection` to avoid the casts at the callsites. https://github.com/dotnet/corefx/blob/c20fcbec2832e52c86b008db89617e2d879bea29/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs#L247-L250